src/signature_pad.js

```diff
@@ -94,14 +94,31 @@ SignaturePad.prototype.clear = function () {
 
 SignaturePad.prototype.fromDataURL = function (dataUrl) {
   const image = new Image();
-  const ratio = window.devicePixelRatio || 1;
-  const width = this._canvas.width / ratio;
-  const height = this._canvas.height / ratio;
+  const deviceRatio = window.devicePixelRatio || 1;
+  const width = this._canvas.width / deviceRatio;
+  const height = this._canvas.height / deviceRatio;
 
   this._reset();
   image.src = dataUrl;
   image.onload = () => {
-    this._ctx.drawImage(image, 0, 0, width, height);
+    const imgWidth = image.width / deviceRatio;
+    const imgHeight = image.height / deviceRatio;
```

Comment:
This code works for me if imgWidth and imgHeight are not divided by deviceRatio.